### PR TITLE
fix(bazel): surface error message when gazelle cant process glob expression

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Closes DINF-89

Gazelle sometimes have trouble processing glob expressions, and this isn't reported as a failure even though it ultimately results in the `BUILD.bazel` not being correctly updated.

## Test plan

* Manual testing

In `client/web/BUILD.bazel`, add a new `src` to the `web_lib` ts_project target that includes a glob pattern.

```
...
ts_project(
    name = "web_lib",
    srcs = glob(["!src/playwright/*.spec.ts"]) + [
        "src/Index.tsx",
        "src/LegacyLayout.tsx",
        "src/LegacyRouteContext.tsx",
        "src/LegacySourcegraphWebApp.tsx",
        "src/PageError.tsx",
        "src/SearchQueryStateObserver.tsx",
        "src/SourcegraphWebApp.tsx",
...
```

When you run `go run ./dev/sg bazel configure`, the command should fail with an error message instead of returning exit 0.

### CODE REVIEW

File: dev/ci/bazel-configure.sh [hunk 1]
```diff
@@ -8,12 +8,24 @@ export PATH
...
-bazel \
-  --bazelrc=.bazelrc \
-  --bazelrc=.aspect/bazelrc/ci.bazelrc \
-  --bazelrc=.aspect/bazelrc/ci.sourcegraph.bazelrc \
-  run //:gazelle
+# TO enable us access the error message / warning returned by gazelle, we trap stderr in a variable
+# so we can check for glob warnings and report accordingly.
+stderr_output=$(bazel \
+    --bazelrc=.bazelrc \
+    --bazelrc=.aspect/bazelrc/ci.bazelrc \
```
Comment (Strum355) [inline dev/ci/bazel-configure.sh:15]: We should probably only include these two ci bazelrc if $CI is set

Comment (BolajiOlajide) [reply dev/ci/bazel-configure.sh]: Updated.

File: dev/ci/bazel-configure.sh [hunk 2]
```diff
@@ -8,12 +8,27 @@ export PATH
...
-  --bazelrc=.aspect/bazelrc/ci.sourcegraph.bazelrc \
-  run //:gazelle
+bazelArgs=("--bazelrc=.bazelrc")
+
+if [ "${CI:-}" ]; then
+  bazelArgs+=("--bazelrc=.aspect/bazelrc/ci.bazelrc")
+  bazelArgs+=("--bazelrc=.aspect/bazelrc/ci.sourcegraph.bazelrc")
+fi
+
+# TO enable us access the error message / warning returned by gazelle, we trap stderr in a variable
```
Comment (jhchabran) [inline dev/ci/bazel-configure.sh:18]:
```suggestion
# To enable us access the error message / warning returned by gazelle, we trap stderr in a variable
```

File: dev/ci/bazel-configure.sh [hunk 3]
```diff
@@ -8,12 +8,27 @@ export PATH
...
+stderr_output=$(bazel "${bazelArgs[@]}" run //:gazelle 2>&1 >/dev/null)
+
+# If the messages output to stderr includes `could not merge expression`, then it means gazelle
+# encountered an issue while reading a glob expression. We surface that to the user so they can
+# fix.
+if echo "${stderr_output}" | grep -q "could not merge expression"; then
+  echo "${stderr_output}"
+
+  gazelle_err_line=$(echo "${stderr_output}" | grep -m 1 -o '^gazelle:.*')
+  echo "gazelle encountered an issue processing glob expression, the BUILD file is not updated. ${gazelle_err_line}"
```
Comment (jhchabran) [inline dev/ci/bazel-configure.sh:29]: nice touch 🙏

**Review by jhchabran — APPROVED** [review]

**Pull request merged**